### PR TITLE
Sparse Softmax Cross Entropy

### PR DESCRIPTION
**Pull request by lilleswing**
So you don't have to one_hot encode in the feed dict generator forcing you to write your own feed dict generator for classification.

### CODE REVIEW

**Comment by rbharath** [comment]
LGTM.

Btw, adding on these tricks to one of our IPython notebooks would be great!
